backend/src/util.rs

```diff
@@ -0,0 +1,37 @@
+use load::Data;
+use chrono::NaiveDateTime;
+
+/// Return where the passed date is (Ok(usize)) or should go (Err(usize)) in
+/// the data slice.
+fn get_insert_location(data: &[Data], date: &NaiveDateTime) -> ::std::result::Result<usize, usize> {
```

Comment:
I was tempted, but `find` [to me] implies returning `&Item` (`&Data`) in this case; `position` could work, but since this can return the position where something should go, not where it is, I wasn't too sure.

Also, do you think it makes sense to return Result<usize, usize> here? Or would returning usize and doing the if is_ok(), unwrap(), else, unwrap_err() logic in here be better? We perform that logic anyway below near as I can tell.